src/state.py

```diff
@@ -155,6 +156,40 @@ def _get_agent_meta_map_from_relation(
     return {unit.name: AgentMeta.from_agent_relation(relation.data[unit]) for unit in remote_units}
 
 
+class ProxyConfig(BaseModel):
+    """Configuration for accessing Jenkins through proxy.
+
+    Attributes:
+        http_proxy: The http proxy URL.
+        https_proxy: The https proxy URL.
+        no_proxy: Comma separated list of hostnames to bypass proxy.
+    """
+
+    http_proxy: typing.Optional[HttpUrl]
```

Comment:
Wouldn't AnyHttpUrl be more suitable?

Reply:
Fixed, thank you!

Reply:
I remembered the reason for using `HttpUrl` over `AnyHttpUrl` writing the unit tests.
`AnyHttpUrl` wouldn't automatically determine the port from the scheme (http / https) whereas `HttpUrl` would give me 80 / 443. I need the ports to be automatically inferred and hence used the HttpUrl. Would this be okay? Otherwise I would need to write a custom parser.

Reply:
Sure. Thanks for clarifying